apps/web/src/store/use-env-controller.ts

```diff
@@ -0,0 +1,65 @@
+import { useQuery, useQueryClient } from 'react-query';
+import { IEnvironment } from '@novu/shared';
+import { getCurrentEnvironment, getMyEnvironments } from '../api/environment';
+import { useContext, useEffect, useState } from 'react';
+import { api } from '../api/api.client';
+import { AuthContext } from './authContext';
+import { QueryKeys } from '../api/query.keys';
+
+export type EnvironmentContext = {
+  readonly: boolean;
+  isLoading: boolean;
+  environment: IEnvironment | undefined;
+  setEnvironment: (environment: string) => void;
+};
+
+export const useEnvController = (): EnvironmentContext => {
+  const queryClient = useQueryClient();
+  const { setToken } = useContext(AuthContext);
+  const [readonly, setReadonly] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const { data: environments, isLoading: isLoadingMyEnvironments } = useQuery<IEnvironment[]>(
+    QueryKeys.myEnvironments,
+    getMyEnvironments
+  );
+  const { data: environment, isLoading: isLoadingCurrentEnvironment } = useQuery<IEnvironment>(
+    QueryKeys.currentEnvironment,
+    getCurrentEnvironment
+  );
+
+  useEffect(() => {
+    if (!environment) {
+      return;
+    }
+    setReadonly(environment.name === 'Production');
```

Comment:
Can we test here for not existing parentId instead? In this case it will be future proofed for when we add multiple environments?